### PR TITLE
Include rpi-eeprom repo as a git submodule

### PR DESCRIPTION
**Pull request by timg236**
* Avoid duplicated binaries - it also makes it easier for users to select a different EEPROM binary i.e. they just pick a different submodule version
* Preliminary work for moving the rpi-bootloader-key-convert and rpi-otp-private-key into rpi-eeprom so that we can create an raspberrypi-secure-boot buildroot package.

### CODE REVIEW

**Comment by timg236** [comment]
@XECDesign @peterharperuk

File: Readme.md [hunk 1]
```diff
@@ -23,11 +23,13 @@ For more information run `rpiboot -h`.
...
 
-**This git repository uses symlinks. For Windows builds clone the repository under Cygwin**
+* This git repository uses symlinks. For Windows builds clone the repository under Cygwin.
+* Instead of duplicating the EEPROM binaries and tools the rpi-eeprom repository
+  is included as a (git submodule)[https://git-scm.com/book/en/v2/Git-Tools-Submodules]
 
 ```
 sudo apt install git libusb-1.0-0-dev pkg-config
-git clone --depth=1 https://github.com/raspberrypi/usbboot
+git clone --recurse-submodules --depth=1 https://github.com/raspberrypi/usbboot
```
Comment (peterharperuk) [inline Readme.md:32]: Does --depth=1 apply to submodules or do you have to include --shallow-submodules ?

File: tools/update-pieeprom.sh [hunk 2]
```diff
@@ -137,6 +137,12 @@ while getopts "c:hi:o:k:p:" option; do
     esac
 done
 
+submodule_check="${script_dir}/rpi-eeprom/firmware/stable/recovery.bin"
+if [ ! -f "${submodule_check}" ]; then
+   echo "WARNING:${submodule_check} not found. To update the rpi-eeprom submodule run:"
+   echo "git submodule init && git submoule update"
```
Comment (pelwell) [inline tools/update-pieeprom.sh:143]: `submoule` - not so mussely?

File: Readme.md (hunk 1 above)
Comment (timg236) [reply Readme.md:32]: --depth=1 applies to 'update' but I think you're right about the clone command. It needs --shallow-submodules to pass --depth=1 to the submodules if --recurse-submodules is specified.

**Comment by timg236** [comment]
Seems to work but will wait until after bumping the rpi-eeprom default release next week before merging.

**Comment by timg236** [comment]
@peterharperuk @pelwell @tdewey-rpi

Reworked this change now that Pi5 has been released in order to stop duplicating the rpi-eeprom tools. Currently, the EEPROM binary is still duplicated but that can be tidied up in a later commit.

**Comment by peterharperuk** [comment]
Looks ok

**Pull request merged**